### ISSUE TITLE
safer treesitter fold?

### ISSUE THREAD

**Issue by GopherJ** [neovim/neovim#30790]
### Problem

vim.treesitter.foldexpr can throw error easily and I don't know how to reproduce it yet, the following safer_fold doesn't work either, anyway to avoid throwing out errors?

```lua
function _G.safe_foldexpr()
	local ok, result = pcall(vim.treesitter.foldexpr)
	if ok then
		return result
	else
		return "0"
	end
end

vim.wo.foldmethod = "expr"
vim.wo.foldexpr = "v:lua.safe_foldexpr()"
vim.o.foldlevel = 99
```

```
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/treesitter.lua:97: There is no parser available for buffer 44 and one could not be created because lang could not be determined. Either pass l
ang or set the buffer filetype
stack traceback:
        [C]: in function 'error'
        /usr/share/nvim/runtime/lua/vim/treesitter.lua:97: in function 'get_parser'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:117: in function 'compute_folds_levels'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:317: in function 'fn'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:293: in function </usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:289>
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/treesitter.lua:97: There is no parser available for buffer 44 and one could not be created because lang could not be determined. Either pass l
ang or set the buffer filetype
stack traceback:
        [C]: in function 'error'
        /usr/share/nvim/runtime/lua/vim/treesitter.lua:97: in function 'get_parser'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:117: in function 'compute_folds_levels'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:317: in function 'fn'
        /usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:293: in function </usr/share/nvim/runtime/lua/vim/treesitter/_fold.lua:289>


```

### Steps to reproduce

```lua
require("nvim-treesitter.configs").setup({ fold = { enable = true }})
function _G.safe_foldexpr()
	local ok, result = pcall(vim.treesitter.foldexpr)
	if ok then
		return result
	else
		return "0"
	end
end

vim.wo.foldmethod = "expr"
vim.wo.foldexpr = "v:lua.safe_foldexpr()"
vim.o.foldlevel = 99
```

lsp reference with nvim-bqf can throw out this error but I don't know how to reproduce it yet

### Expected behavior

safe fold

### Nvim version (nvim -v)

NVIM v0.10.1 Build type: Debug LuaJIT 2.1.1713484068 Run "nvim -V1 -v" for more info

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Linux Mint 22.04

### Terminal name/version

Alacrittyalacritty 0.13.2 (bb8ea18e)

### $TERM environment variable

tmux-256color

### Installation

apt

**Comment by GopherJ** [comment]
related to: https://github.com/neovim/neovim/issues/23423

**Comment by clason** [comment]
> require("nvim-treesitter.configs").setup({ fold = { enable = true }})

Do not mix different foldexpr providers. And only enable the foldexpr for specific filetypes where you know you have a parser available.

**Comment by GopherJ** [comment]
@clason I do have a parser for the filetypes but when nvim-bqf creates float windows, I think some error triggered, in any case I don't think an error like this should be throw because it influences a lot

**Comment by justinmk** [comment]
Won't this be "fixed" after the next stage of https://github.com/neovim/neovim/pull/30382 , where `get_parser` will no longer throw?

**Comment by GopherJ** [comment]
as of NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1732813678
Run "nvim -V1 -v" for more info


things are still not fixed

**Comment by justinmk** [comment]
> as of NVIM v0.10.2

0.10.2 was released on oct 3, so the bug cannot possibly be fixed on it

**Comment by GopherJ** [comment]
where can I find development release 0.11? that pr is still not included in 0.10.3

**Comment by zeertzjq** [comment]
https://github.com/neovim/neovim/releases
